test/contrib/sinatra/tracer_test_base.rb

```diff
@@ -0,0 +1,17 @@
+
+require 'sinatra/base'
+require 'ddtrace'
+require 'ddtrace/contrib/sinatra/tracer'
+require 'helper'
+require 'rack/test'
+
+class TracerTestBase < Minitest::Test
+  include Rack::Test::Methods
+
+  def setup
+    skip unless ENV['TEST_DATADOG_INTEGRATION'] # requires a running agent
```

Comment:
I'm not sure you need to skip this test if you're using a `FauxWriter`. AFAIK it's indeed designed *not* to use the default tracer, and instead redirect all output to some in-memory structure. So you don't need the agent and it can run without `TEST_DATADOG_INTEGRATION` set.